qhana_plugin_registry/db/models/templates.py

```diff
@@ -267,34 +238,30 @@ class TemplateTab(IdMixin, ExistsMixin, NameDescriptionMixin):
     def plugin_filter(self) -> dict:
         return json.loads(self.filter_string)
 
-    @classmethod
-    def get_by_name(cls, name: str) -> "Optional[TemplateTab]":
-        q = select(cls).filter(cls.name == name)
+    def get_by_name(
+        template: UiTemplate, name: str, location: Optional[str] = None
+    ) -> "Optional[TemplateTab]":
+        q = (
+            select(TemplateTab)
+            .filter(TemplateTab.template == template)
+            .filter(TemplateTab.name == name)
```

Comment:
```suggestion
            .filter(TemplateTab.template == template, TemplateTab.name == name)
```

You can combine multiple filter calls like this (it will create an implicit and).